vtex-sites/nextjs.store
# fix: Page error codes

## What's the purpose of this pull request?
This PR fixes a bug introduced on  #88. Also, fixes one of the TODOs regarding GraphQL status code. 

## How does it work?
#88 fixed wrong 404s on PDPs when an error occurred between `function` -> `ecommerce API`. This fix, however, didn't take into account that users entering wrong slugs should still get a 404, not a 500. This PR fixes all of these problems. 

To way it works is that now it reads the error status thrown inside `@faststore/api` before returning the response. 

## How to test it?
1. Pick a PDP, say `/4k-philips-monitor-99988213/p`
2. When visiting `/4k-philips-monitor-99988213/p` you should get a 200 with the right page being rendered
3. Visiting a non existing page, say ``/4k-philips-monitor/p` should get you a 404 page
4. When an error occurs between function -> ecommerce API, you should get a 500. (To simulate this behavior, run locally and turn off the wifi)

## References
Also, check the reference PR on [@faststore/api](https://github.com/vtex/faststore/pull/1361)
